### PR TITLE
Don't use _disable_current_modes as decorator

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132809
* #132804
* #132802
* #132801

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/freezing.py [hunk 1]
```diff
@@ -147,12 +147,30 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
...
+            ):
+                with torch._dispatch.python.no_python_dispatcher():
+                    e_t = ErasedTensor(tensor, attr_name, mod)
+                if isinstance(tensor, torch.nn.Parameter):
+                    e_t.requires_grad_(True)
+                    e_t._is_param = True  # type: ignore[attr-defined]
+                setattr(mod, attr_name, e_t)
+
 
+def discard_traced_gm_params(mod: torch.fx.GraphModule):
```
Comment (albanD) [inline torch/_inductor/freezing.py:172]: What is the difference between this one and invalidate_eager_modules above?
Given how confused git diff is, they are litteratlly the same with an extra for loop in the top one. Sohuld we just have one call the other?

**Review by albanD — APPROVED** [review]
optional cleanup, SGTM otherwise

File: torch/_inductor/freezing.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/freezing.py:172]: Yeah, it does seem like it. When you ignore whitespace you get the "normal" diff lol

**Pull request closed without merging**